code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -271,115 +266,139 @@
+				emote_see = list("hunts for treasure.","stares coldly...","gnashes his tiny corgi teeth!")
+				emote_hear = list("growls ferociously!", "snarls.")
+				speak = list("Arrrrgh!!","Grrrrrr!")
+
+			if(/obj/item/clothing/head/ushanka)
+				name = "[pick("Comrade","Commissar","Glorious Leader")] [real_name]"
+				desc = "A follower of Karl Barx."
+				emote_see = list("contemplates the failings of the capitalist economic model.", "ponders the pros and cons of vanguardism.")
+
+			if(/obj/item/clothing/head/warden, /obj/item/clothing/head/collectable/police)
+				name = "Officer [real_name]"
+				emote_see = list("drools.","looks for donuts.")
+				desc = "Stop right there criminal scum!"
+
+			if(/obj/item/clothing/head/wizard/fake,	/obj/item/clothing/head/wizard,	/obj/item/clothing/head/collectable/wizard)
+				name = "Grandwizard [real_name]"
+				speak = list("YAP", "Woof!", "Bark!", "AUUUUUU", "EI  NATH!")
+
+			if(/obj/item/clothing/head/cardborg)
+				name = "Borgi"
+				speak = list("Ping!","Beep!","Woof!")
+				emote_see = list("goes rogue.", "sniffs out non-humans.")
+				desc = "Result of robotics budget cuts."
+
+			if(/obj/item/weapon/bedsheet)
+				name = "\improper Ghost"
+				speak = list("WoooOOOooo~","AUUUUUUUUUUUUUUUUUU")
+				emote_see = list("stumbles around.", "shivers.")
+				emote_hear = list("howls!","groans.")
+				desc = "Spooky!"
+
+			if(/obj/item/clothing/head/helmet/space/santahat)
+				name = "Santa's Corgi Helper"
+				emote_hear = list("barks Christmas songs.", "yaps merrily!")
+				emote_see = list("looks for presents.", "checks his list.")
+				desc = "He's very fond of milk and cookies."
+
+			if(/obj/item/clothing/head/soft)
+				name = "Corgi Tech [real_name]"
+				desc = "The reason your yellow gloves have chew-marks."
+
+			if(/obj/item/clothing/head/hardhat/reindeer)
+				name = "[real_name] the red-nosed Corgi"
+				emote_hear = list("lights the way!", "illuminates.", "yaps!")
+				desc = "He has a very shiny nose."
+
+			if(/obj/item/clothing/head/sombrero)
+				name = "Segnor [real_name]"
+				desc = "You must respect elder [real_name]"
+
+			if(/obj/item/clothing/head/hopcap)
+				name = "Lieutenant [real_name]"
+				desc = "Can actually be trusted to not run off on his own."
+
+			if(/obj/item/clothing/head/helmet/space/hardsuit/deathsquad)
+				name = "Trooper [real_name]"
+				desc = "That's not red paint. That's real corgi blood."
+
+			if(/obj/item/clothing/mask/gas/clown_hat)
+				name = "[real_name] the Clown"
+				desc = "Honkman's best friend."
+				speak = list("HONK!", "Honk!")
+				emote_see = list("plays tricks.", "slips.")
+			else
+				special_hat = 0
+	
+	var/special_back = 0
+	if(inventory_back)
+		special_back = 1
+		switch(inventory_back.type)
+			if(/obj/item/clothing/suit/space/hardsuit/deathsquad)
+				name = "Trooper [real_name]"
+				desc = "That's not red paint. That's real corgi blood."
+			else
+				special_back = 0
+
+	if(!special_hat && !special_back)
```

Comment:
@KazeEspada except kaze it triggers if they have ANY HAT or ANY BACK, so they're all special, and thus the extra vars are unneeded.